server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/MultipleDatabaseChangeRuntimeFlowableTask.java

```diff
@@ -75,16 +81,48 @@ public class MultipleDatabaseChangeRuntimeFlowableTask extends BaseODCFlowTaskDe
     private FlowInstanceRepository flowInstanceRepository;
     @Autowired
     private ServiceTaskInstanceRepository serviceTaskInstanceRepository;
+    @Autowired
+    private EntityManager entityManager;
 
     @Override
     public boolean cancel(boolean mayInterruptIfRunning, Long taskId, TaskService taskService) {
-        // todo implement later
-        throw new UnsupportedOperationException();
+        // used to interrupt the start method
+        this.isAbortedForStart = true;
+        while (System.currentTimeMillis() - getStartTimeMilliSeconds() <= getTimeoutMillis()) {
+            if (isFinished) {
+                break;
+            }
+            if (isContinuedForCancel) {
+                if (this.flowInstanceIds == null || this.flowInstanceIds.isEmpty()) {
+                    break;
+                }
+                this.flowInstanceIds = listNeedCancelIds(this.flowInstanceIds);
+                for (Long flowInstanceId : this.flowInstanceIds) {
+                    try {
+                        this.flowInstanceService.cancelSubFlowInstance(flowInstanceId);
```

Comment:
it may take long time if there exists many sub flow intances, may involve thread pool.

Reply:
ok